src/main/scala/nodes/learning/KernelGenerator.scala

```diff
@@ -22,9 +22,8 @@ abstract class KernelGenerator extends Transformer[DenseVector[Double], DenseVec
   /* Cached KernelBlockBlock (used for training), and corresponding
    * blockIdx
    */
-  var cachedBlockBlock:Option[(DenseMatrix[Double], Seq[Int])] = None
 
-  def apply(testMat: RDD[DenseVector[Double]], blockIdxs: Seq[Int], same: Boolean = false, cache: Boolean = false): RDD[DenseVector[Double]]
+  def apply(testMat: RDD[DenseVector[Double]], blockIdxs: Seq[Int], same: Boolean = false): RDD[DenseVector[Double]]
```

Comment:
yeah so the KernelMatrix can still take a function (it can even the same generator object) in the constructor.  But the nice thing about what I said before is that it will give us some modularity where we can say

`val kernelRidgePipeline = new GaussianKernelGenerator() andThen new LinearSystemSolver()` and `kernelRidgePipeline` can be `fit` and then applied to test data etc. i.e. matching the usual keystone workflow. The only difference in this case is that the intermediate datastructure is a `KernelMatrix` instead of `RDD[DenseVector[Double]]` and thus we also use the type system effectively. i.e. if you did `andThen new BlockWeightedLeastSquares` it will fail.